beluga_tutorial/README.md

```diff
@@ -0,0 +1,505 @@
+
+### 3.7 Resample
+The updated belief is prefigured by the spatial distribution of the particles $x_{t-1}$ in the set and their importance weights. A few of the particles will have migrated to state regions with low probability, however, and their importance weights will therefore be low.
+To correct this, the update step is completed by performing a **resampling process**, which consist of drawing a new set of particles $x_t$ from the current set, with replacement, using importance weights as unnormalized probabilities. This process causes particles with low weights to be discarded and particles with high weights to be propagated multiple times into the new particle set.
+
+Adaptive Monte Carlo Localization (AMCL) uses different sampling techniques to dynamically adjust the number of particles to match the complexity of the posterior distribution. For the simplicity of this tutorial, we'll keep a fixed number of particles, as the performance is adequate.
+
+```cpp
+// Resample
+particles |= beluga::views::sample | ranges::views::take_exactly(tutorial_params.number_of_particles) |
+              beluga::actions::assign;
+```
+* `beluga::views::sample` implements a multinomial resampling on `particles`.
+* `ranges::views::take_exactly(tutorial_params.number_of_particles)` defines the number of particles to take, using the **number_of_particles** parameter.
+* `beluga::actions::assign` effectively converts any view into an action and assigns the result to `particles`.
+
+## 4 Compile and run code
+### 4.1 package.xml
+Create a `package.xml` file and put the following inside:
+
+```xml
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>beluga_tutorial</name>
+  <version>0.0.0</version>
+  <description>TODO: Package description</description>
+  <maintainer email="user@mail.com">developer</maintainer>
+  <license>Apache License 2.0</license>
+
+  <buildtool_depend>cmake</buildtool_depend>
+
+  <depend>beluga</depend>
+  <depend>yaml_cpp_vendor</depend>
+
+  <export>
+    <build_type>cmake</build_type>
+  </export>
+</package>
+```
+
+This declares the package needs `beluga` and `yaml_cpp` when its code is built and executed.
+
+### 4.2 CMakeLists.txt
+The final `CMakeLists.txt` looks like:
+
+```cmake
+cmake_minimum_required(VERSION 3.16)
+
+project(beluga_tutorial)
+
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+if(NOT CMAKE_BUILD_TYPE)
+  message(STATUS "Setting build type to 'Release' as none was specified.")
+  set(CMAKE_BUILD_TYPE
+      "Release"
+      CACHE STRING "Build type" FORCE)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(
+    -Wall
+    -Wconversion
+    -Wextra
+    -Werror
+    -Wpedantic)
+endif()
```

Comment:
@Alondruck nit: this seems like unnecessary clutter for a tutorial. Consider removing it.